source/extensions/filters/http/lua/lua_filter.h

```diff
@@ -285,6 +297,37 @@ class StreamHandleWrapper : public Filters::Common::Lua::BaseLuaObject<StreamHan
   State state_{State::Running};
   std::function<void()> yield_callback_;
   Http::AsyncClient::Request* http_request_{};
+  FireAndForgetHttpWriter* fireAndForgetHttpWriter_;
+};
+
+/**
+ * HTTP client that performs non-blocking HTTP calls and ignores the response.
+ */
+class FireAndForgetHttpWriter : public Filters::Common::Lua::BaseLuaObject<FireAndForgetHttpWriter>,
+                                public Http::AsyncClient::Callbacks {
+public:
+  FireAndForgetHttpWriter(Filter& filter);
+
+  int luaHttpCallNonblocking(lua_State* state);
+
+  // Http::AsyncClient::Callbacks
+  void onSuccess(Http::ResponseMessagePtr&&) override {}
+
+  void onFailure(Http::AsyncClient::FailureReason) override {}
+
+private:
+  Filter& filter_;
+};
+
+/**
+ * A class with shared code for building and making HTTP calls.
+ */
+class LuaFilterUtil : public Filters::Common::Lua::BaseLuaObject<LuaFilterUtil> {
```

Comment:
Instead of making these utility functions exposed outside of this class/translation unit i'd suggest putting them in the anonymous namespace in the .cc file. This tends to be the preferred way of doing helper functions that are only used in one .cc file and that doesn't benefit from being a member function

you can look for `namespace {` in existing code to see what an anonymous namespace looks like